tests/net/host.cpp

```diff
@@ -0,0 +1,44 @@
+#include <gtest/gtest.h>
+#include <net/utilities.hpp>
+#include <net.hpp>
+#include <iostream>
+
+TEST(Host, LocalHost) { // ~15ms
+    auto localhost = arch::net::Host::localhost(true);
+
+    EXPECT_EQ(localhost.ip(), arch::net::IPv4::localhost);
+
+    char buffer[256]{};
+    gethostname(buffer, 256);
+
+    EXPECT_EQ(strcmp(localhost.hostname().c_str(), buffer), 0);
+}
+
+TEST(Host, MyIPs) { // ~3ms
+    std::vector<arch::net::IPv4> expected_ips{
+        // insert your IPs here
```

Comment:
Unit tests should be self-sufficient and not require any changes in code in order to run. If it's impossible in this test, then it doesn't belong here.